### ISSUE TITLE
Proposal: check use, import, alias, require having same order across files

### ISSUE THREAD

**Issue by progsmile** [rrrene/credo#540]
Hi there! Thanks for package!
I didn't find rule to check order for directives: use, import, alias, require 
Wish it would be, to make code more consistent and prevent human mistakes
In future we can add option to set particular order, like `{Credo.Check.Readability.ModuleDirectivesOrder, order: [:use, :import, :require, :alias]}`

So, if you don't mind I'll try to implement this and do pull request

Here are examples:
- Valid
`use Ecto.Schema`
`import Ecto.Query`
`alias Ecto.Paging`
`require Logger`

- Also valid 
`import Ecto.Query`
`require Logger`

- Invalid
`require Logger`
`import Ecto.Query`
`alias Ecto.Paging`

What do you think?)

**Comment by progsmile** [comment]
Does anybody consider it helpful?

**Comment by rrrene** [comment]
Thx for proposing this! :+1:

I thought about adding a check like this, but then I thought "Is there one true order of directives? Isn't there a situation where I want to first `alias` and then `use`?". 

Why is it this order in your proposal?

**Comment by progsmile** [comment]
Actually it's from [this style guide](https://github.com/christopheradams/elixir_style_guide#module-attribute-ordering) and in our team we do so, but still have inconsistent files. Personally I would choose `use`, `import`, `require`, `alias` sequence as default, in order directive impacts module.
As I wrote above it would support option `order`, so any order can be set )

**Comment by rrrene** [comment]
You are right. 

One note: Let's use `use/import/alias/require`, since that's also in the Elixir Core Team's style guide. :+1:

**Comment by progsmile** [comment]
Ok, I’ll do pull request soon

**Comment by janpieper** [comment]
Had the same idea a few minutes ago :rofl:

Would be great to have such a check in Credo! :+1:

**Comment by progsmile** [comment]
Started with nested modules case, so need to combine modules and their directives. Now I use `Credo.Code.postwalk`, but wonder if is there more appropriate function?

```elixir
defmodule CredoSampleModule do
  use TestUse
  require TestRequire

  defmodule CredoSampleModule.Nested do
    use Nested.Use
    require Nested.Require

    defmodule CredoSampleModule.Nested.NestedTwo do
      use Nested.Second.Use, opts: []
      require Nested.Second.Requre
    end
  end
end
```

**Comment by progsmile** [comment]
Suppose such cases are quire rare (nested modules up to 3rd level), but still want handle this )

**Comment by bamorim** [comment]
I did some checks on what Credo was doing according [this style guide](https://github.com/christopheradams/elixir_style_guide#module-attribute-ordering) and this was one of the things credo was not checking.

The module ordering was one of the issues. Are you planning to go with the "full ordering" from [elixir-style-guide](https://github.com/christopheradams/elixir_style_guide#module-attribute-ordering) or just `use/import/require/alias`?

I think that including `defstruct/@type/@callback/etc` may be nice as well.

How can I help with that?

**Comment by progsmile** [comment]
@bamorim That would be really cool because of completeness!)
Actually the problem I faced was dealing with nested module - [comment above](https://github.com/rrrene/credo/issues/540#issuecomment-389407113)
If you have solution please do pr )

**Comment by whatyouhide** [comment]
For now, I suggest starting with just `use`/`import`/`alias`/`require`. The reason for that ordering is that they are from most "dangerous" (`use` can do many hidden things) to least dangerous.

**Comment by doomspork** [comment]
@progsmile / @bamorim have either of you started on this?

**Comment by progsmile** [comment]
@doomspork started and not finished, you may close

**Comment by doomspork** [comment]
@progsmile / @bamorim I'd love to get your thoughts on the PR I opened: https://github.com/rrrene/credo/pull/588
